apps/dashboard/src/components/workflow-editor/workflow-canvas.tsx

```diff
@@ -57,6 +56,17 @@ const panOnDrag = [1, 2];
 // y distance = node height + space between nodes
 const Y_DISTANCE = NODE_HEIGHT + 50;
 
+const mapStepToNodeContent = (step: Step): string => {
+  const controlValues = step.controls.values;
+
+  switch (step.type) {
+    case StepTypeEnum.DELAY:
+      return `Delay for ${controlValues.amount} ${controlValues.unit}`;
+    default:
+      return 'Unknown step type';
+  }
+};
+
```

Comment:
I can imagine something similar for digest but we need to think about the copy, especially the cron/scheduled one.